service/src/main/java/bio/terra/workspace/service/resource/ValidationUtils.java

```diff
@@ -41,12 +42,47 @@
   public static final Pattern RESOURCE_NAME_VALIDATION_PATTERN =
       Pattern.compile("^[a-zA-Z0-9][-_a-zA-Z0-9]{0,1023}$");
 
+  private static final String GOOG_PREFIX = "goog";
+  private static final ImmutableList<String> GOOGLE_NAMES = ImmutableList.of("google", "g00gle");
+
+  /**
+   * Validates gcs-bucket name following Google documentation
+   * https://cloud.google.com/storage/docs/naming-buckets#requirements on a best-effort base.
+   *
+   * This method DOES NOT guarentee that the bucket name is valid.
+   * @param name gcs-bucket name
+   */
   public static void validateBucketName(String name) {
     if (StringUtils.isEmpty(name) || !BUCKET_NAME_VALIDATION_PATTERN.matcher(name).matches()) {
       logger.warn("Invalid bucket name {}", name);
       throw new InvalidReferenceException(
```

Comment:
I think we set a bad precedent here before, but this (and the other cases below) shouldn't be a `InvalidReferenceException` since this isn't just validating references. 
Could you make a new exception instead (something like `InvalidCloudSpecification`, status code 400)?

Reply:
there's an existing InvalidNameException. Wdyt I just use that?

Reply:
sounds good for the exceptions here, but the 400 in `CreateGcsBucketStep` can also be caused by other fields so that wouldn't be right there. 
that one already has the error from GCP included though, so it could just be a more generic `BadRequestException`